IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -502,7 +507,7 @@ - (MSIDRefreshToken *)getRTForAccount:(MSIDAccount *)account
     }
     
     // If token wasn't found and legacy user ID is available, try to look by legacy user id
-    if (!refreshToken && [NSString msidIsStringNilOrBlank:account.legacyUserId])
+    if (!refreshToken && ![NSString msidIsStringNilOrBlank:account.legacyUserId])
```

Comment:
Can you explain why you changed it to `![NSString msidIsStringNilOrBlank:account.legacyUserId]`?

Reply:
Yes, we want to look for token by legacy user ID as a fallback, if no tokens were found by normal user ID. For that legacyUserId needs to be present. Before there was a bug that we wanted to get tokens with a legacy user ID, but were checking for legacy user ID to be empty. The bug got revealed with an additional integration test I added and I fixed the bug.